code/modules/surgery/organs/lungs.dm

```diff
@@ -267,10 +269,13 @@
 				H.Sleeping(min(H.AmountSleeping() + 100, 200))
 		else if(SA_pp > 0.01)	// There is sleeping gas in their lungs, but only a little, so give them a bit of a warning
 			if(prob(20))
+				n2o_euphoria = EUPHORIA_ACTIVE
 				H.emote(pick("giggle", "laugh"))
-				SEND_SIGNAL(owner, COMSIG_ADD_MOOD_EVENT, "chemical_euphoria", /datum/mood_event/chemical_euphoria)
+			else
+				n2o_euphoria = EUPHORIA_LAST_FLAG
```

Comment:
You could make this the default value and avoid reasserting it at all

Reply:
aaand done